src/resolvers/dockerfile.ts

```diff
@@ -15,15 +30,21 @@ export class DockerfileResolver implements Resolver {
 		this.dockerfileContents = file.contents.toString();
 	}
 
-	public needsEntry(filePath: string): boolean {
-		return path.basename(filePath) === 'Dockerfile';
+	public needsEntry(
+		entry: TarEntryPath,
+		specifiedDockerfilePath?: string,
+	): boolean {
+		return entry.path === (specifiedDockerfilePath || 'Dockerfile');
```

Comment:
Two cases:
* If a specifiedDockeriflePath was specified, say `'service/MyDockerfile'`, then this resolver (`DockerfileResolver `) will only match a tar entry whose full path is identical to that (`entry.path === specifiedDockerfilePath`).
* Otherwise, it will match `Dockerfile` at the root of the project directory tree, as `entry.path` is the full tar entry path which must be exactly `'Dockerfile'`. This root-only rule was previously enforced in `resolveTarStreamOnEntry()` through a test: `if (path.normalize(name).split(path.sep).length === 1)`. That test had been added by a recent-ish pull request:
_fix: Correctly resolve contexts with multiple bundle descriptions_
https://github.com/balena-io-modules/resin-bundle-resolve/pull/32/files